Chrysalis-Papilio/Soen-490-Papilio-Android
# App fixes 1

# General info

**Issue number**: NA

**Task description**: 

 - Added ability to upload profile pic to backend
 - Added more unit tests for FavoriteActivitiesVM, UpcomingActivitiesVM and UserProfileVM
 - Added and improved error handling for various functions in the UserRepository and the ActivityRepository

# Testing
To test this feature you have to:

1. Go to the User Profile Page
2. Change your DP by choosing any photo and following on-screen instructions
3. Verify that your DP updates on the profile page
4. Now go back to the previous page and come back to the user profile page again.
5. Verify that the new DP you uploaded is still there
6. If steps 3 and 5 are verfied successfully, this PR is therefore complete.
